### PR TITLE
fix/ssnlist

### PR DESCRIPTION
**Pull request by renlulu**
This PR is to:
1. Remove verbose log within `WithdrawalStakeRewards`, basically two lists used for debug purpose.
2. Add an additional check to make people pass through withdraw stake amount if there is no rewards to be withdrawal.
3. Event log for withdraw stake amount
4. Delete map `deleg_stake_per_cycle` in procedure `TruncateDeleg`
5. Update readme.

### CODE REVIEW

**Review by bb111189 — APPROVED** [review]

File: contracts/ssnlist.scilla [hunk 1]
```diff
@@ -914,11 +919,10 @@ transition ChangeBNumReq(input_bnum_req: Uint128, initiator: ByStr20)
   bnum_req := input_bnum_req
 end
 
-transition UpdateContractAddr(proxy_addr: ByStr20, gzil_addr: ByStr20, initiator: ByStr20)
+transition UpdateGzilAddr(gzil_addr: ByStr20, initiator: ByStr20)
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:922]: Why keep proxy_address as an immutable parameter then?

**Review by AmritKumar — COMMENTED** [review]
Update README

File: contracts/ssnlist.scilla (hunk 1 above)
Comment (renlulu) [reply contracts/ssnlist.scilla:921]: Updated to inmutable parameter.

File: contracts/README.md [hunk 2]
```diff
@@ -344,7 +344,7 @@ parameter `initiator` for the `SSNList` contract.
 |`UpdateVerifierRewardAddr(addr: ByStr20)` | `UpdateVerifierRewardAddr(addr: ByStr20, initiator : ByStr20)`|
 |`UpdateStakingParameters(min_stake: Uint128, min_deleg_stake: Uint128, max_comm_change_rate: Uint128)` | `UpdateStakingParameters(min_stake: Uint128, min_deleg_stake: Uint128, max_comm_change_rate: Uint128, initiator : ByStr20) `|
 |`ChangeBNumReq(input_bnum_req: Uint128)` | ` ChangeBNumReq(input_bnum_req: Uint128, initiator : ByStr20)`|
-|`UpdateContractAddr(proxy_address: ByStr20, gzil_address: ByStr20)` | `UpdateContractAddr(proxy_addr: ByStr20, gzil_addr: ByStr20, initiator : ByStr20)`|
+|`UpdateGzilAddr(gzil_address: ByStr20)` | `UpdateGzilAddr(gzil_addr: ByStr20, initiator : ByStr20)`|
```
Comment (AmritKumar) [inline contracts/README.md:347]: Can we make the parameter name `gzil_addr` consistent across Proxy and SSNList?

File: contracts/ssnlist.scilla [hunk 3]
```diff
@@ -409,7 +409,7 @@ end
 
 (* Check if the caller is the proxy *)
 procedure IsProxy()
-  proxy_tmp <- proxyaddr;
+  proxy_tmp = init_proxy_address;
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:413]:
```suggestion
```

File: contracts/README.md (hunk 2 above)
Comment (renlulu) [reply contracts/README.md:347]: Yes, updated!

File: contracts/ssnlist.scilla (hunk 3 above)
Comment (AmritKumar) [inline contracts/ssnlist.scilla:413]: Why copy in the tmp variable? Now that it is not a mutable variable, you can directly call the builtin.

Comment (renlulu) [reply contracts/ssnlist.scilla:413]: Cool!

Comment (renlulu) [reply contracts/ssnlist.scilla]: Change pushed

**Review by AmritKumar — APPROVED** [review]

**Pull request merged**